utils/doclint/generateDotnetApi.js

```diff
@@ -314,19 +314,37 @@ function generateNameDefault(member, name, t, parent) {
   let enumName = generateEnumNameIfApplicable(member, name, t, parent);
   if (!enumName && member) {
     if (member.kind === 'method' || member.kind === 'property') {
-      // this should be easy to name... let's call it the same as the argument (eternal optimist)
-      let probableName = `${parent.name}${translateMemberName(``, name, null)}`;
-      let probableType = additionalTypes.get(probableName);
-      if (probableType) {
-        // compare it with what?
-        if (probableType.expression != t.expression) {
-          throw new Error(`Non-matching types with the same name. Panic.`);
+      let names = [
+        parent.name,
+        translateMemberName(``, member.name, null),
+        translateMemberName(``, name, null),
+      ];
+      if (names[2] === names[1])
+        names.pop(); // get rid of duplicates, cheaply
+      let attemptedName = names.pop();
+      let compareType = function (left, right) {
+        if (!left || !right) return false;
+        if (left.expression && right.expression)
+          return left.expression !== right.expression;
+        return JSON.stringify(right.properties) !== JSON.stringify(left.properties);
+      }
+      while (true) {
+        // crude attempt at removing plurality
+        if (attemptedName.endsWith('s'))
```

Comment:
This is brittle, will fail on names like `properties`. Perhaps whitelist known names?

Reply:
It is, to some extent... But also, we only have a few of these that are actually plural, so I can work around with something like 

```
if (attemptedName.endsWith('s') && !attemptedName.endsWith('ies`))
```

:-) 

When we add something that'll break this simple logic, I can always revisit this... It's the same if I have to add a map entry, or a check, right?